pkg/function/backup_data_stats.go

```diff
@@ -0,0 +1,116 @@
+}
+
+var _ kanister.Func = (*BackupDataStatsFunc)(nil)
+
+type BackupDataStatsFunc struct{}
+
+func (*BackupDataStatsFunc) Name() string {
+	return "BackupDataStats"
+}
+
+func backupDataStats(ctx context.Context, cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, backupArtifactPrefix, backupID, jobPrefix string) (map[string]interface{}, error) {
+	options := &kube.PodOptions{
+		Namespace:    namespace,
+		GenerateName: jobPrefix,
+		Image:        kanisterToolsImage,
+		Command:      []string{"sh", "-c", "tail -f /dev/null"},
+	}
+	pr := kube.NewPodRunner(cli, options)
+	podFunc := backupDataStatsPodFunc(cli, tp, namespace, encryptionKey, backupArtifactPrefix, backupID)
+	return pr.Run(ctx, podFunc)
+}
+
+func backupDataStatsPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey string, backupArtifactPrefix, backupID string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+		// Wait for pod to reach running state
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrapf(err, "Failed while waiting for Pod %s to be ready", pod.Name)
+		}
+		pw, err := getPodWriter(cli, ctx, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, tp.Profile)
+		if err != nil {
+			return nil, err
+		}
+		defer cleanUpCredsFile(ctx, pw, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+		cmd := restic.StatsCommandByID(tp.Profile, backupArtifactPrefix, backupID, encryptionKey)
+		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stdout)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stderr)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to get backup stats")
+		}
+		return nil, nil
+	}
+}
+
+func (*BackupDataStatsFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+	var namespace, pod, backupArtifactPrefix, backupID, encryptionKey string
+	var err error
+	if err = Arg(args, BackupDataStatsNamespaceArg, &namespace); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataStatsBackupArtifactPrefixArg, &backupArtifactPrefix); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataBackupID, &backupID); err != nil {
+		return nil, err
+	}
+	if err = OptArg(args, BackupDataStatsEncryptionKeyArg, &encryptionKey, restic.GeneratePassword()); err != nil {
+		return nil, err
+	}
+	// Validate the Profile
+	if err = validateProfile(tp.Profile); err != nil {
+		return nil, errors.Wrapf(err, "Failed to validate Profile")
+	}
+	cli, err := kube.NewClient()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create Kubernetes client")
+	}
+	return backupDataStats(ctx, cli, tp, namespace, pod, backupArtifactPrefix, backupID, encryptionKey)
```

Comment:
@DeepikaDixit I think the parameters are mismatched in the func definition.

Reply:
Ahh you are right! I'll change it